src/vunnel/providers/mariner/parser.py

```diff
@@ -0,0 +1,196 @@
+        if test is None or test.state is None or test.state.state_ref is None:
+            return None
+        return self.states_by_id.get(test.state.state_ref, None)
+
+    def get_object(self, definition: Definition) -> RpminfoObject | None:
+        test = self.get_test(definition)
+        if test is None or test.object_value is None or test.object_value.object_ref is None:
+            return None
+        return self.objects_by_id.get(test.object_value.object_ref, None)
+
+    def make_fixed_in(self, definition: Definition) -> FixedIn | None:
+        state = self.get_state(definition)
+        obj = self.get_object(definition)
+        if state is None or state.evr is None:
+            return None
+        if obj is None or obj.name is None:
+            return None
+        version = state.evr.value
+        if state.evr.operation == LTE:
+            version = "None"  # legacy API needs the string "None" instead of None
+        return FixedIn(Name=obj.name, NamespaceName=self.namespace_name(), VersionFormat="rpm", Version=version)
+
+    def vulnerability_id(self, definition: Definition) -> str | None:
+        if definition.metadata is None or definition.metadata.reference is None or definition.metadata.reference.ref_id is None:
+            return None
+        return definition.metadata.reference.ref_id
+
+    def description(self, definition: Definition) -> str | None:
+        if not definition.metadata:
+            return None
+        return definition.metadata.description
+
+    def vulnerabilities(self) -> Generator[Vulnerability, None, None]:
+        for d in self.definitions:
+            if d.metadata is None or d.metadata.severity is None:
+                self.logger.warning("skipping definition because severity could not be found")
+                continue
+            if d.metadata.description:
+                pass
+            link = ""
+            if d.metadata.reference and d.metadata.reference.ref_url:
+                link = d.metadata.reference.ref_url
+            fixed_in = self.make_fixed_in(d)
+            if not fixed_in:
+                continue
+            vulnerability_id = self.vulnerability_id(d)
+            if not vulnerability_id:
+                continue
+            yield Vulnerability(
+                Name=vulnerability_id,  # intentional; legacy API uses Name as field for vulnerability ID.
+                NamespaceName=self.namespace_name(),
+                Description=self.description(d) or "",
+                Severity=d.metadata.severity,
+                Link=link,
+                CVSS=[],  # NOTE: this is not present in OVAL XML.
+                # This is a list because because there are several versions
+                # Each version is a vector of info about the vulnerability
+                # serialized as a string like CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H
+                # TODO: add test image to https://github.com/anchore/test-images
```

Comment:
is this TODO current? Also, why add to the test_images repo? (I think I'm missing something is all)

Reply:
Are we ok having the quality tests depend on a digest in a repo we're not hosting? I think I meant that I should build and push an image built from mariner to use in quality tests.

Reply:
> Are we ok having the quality tests depend on a digest in a repo we're not hosting

indeed, we already have precedence for this too. So no need to capture a separate image in `test_images` unless you're trying to make the image / label set smaller.